src/main/java/de/tum/in/www1/artemis/web/rest/TextAssessmentResource.java

```diff
@@ -189,7 +191,8 @@ public TextAssessmentResource(AuthorizationCheckService authCheckService, TextAs
                     "feedbackReferenceTooLong");
         }
         final TextExercise exercise = textExerciseRepository.findByIdElseThrow(exerciseId);
-        final TextSubmission textSubmission = textSubmissionRepository.getTextSubmissionWithResultAndTextBlocksAndFeedbackByResultId(resultId);
+        checkAuthorization(exercise, null);
```

Comment:
Is it intentional that you use `null` as the user?

Reply:
If you pass `null` the method itself will fetch the user.